### PR TITLE
libpod: improve heuristic to detect cgroup

### PR DESCRIPTION
**Pull request by giuseppe**
improve the heuristic to detect the scope that was created for the container.

This is necessary with systemd running as PID 1, since it moves itself to a different sub-cgroup, thus stats would not account for other processes in the same container.

Closes: https://github.com/containers/podman/issues/12400

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Review by flouthoc — COMMENTED** [review]
`cgroupPath` has a early check to fail if container is not running but `inspect` can be performed on a container which is not running. Here: https://github.com/containers/podman/blob/main/libpod/container.go#L921

Maybe we could remove that check and return empty string if container is not running.

**Comment by giuseppe** [comment]
> Maybe we could remove that check and return empty string if container is not running.

yes good idea, fixed and pushed a new version

**Review by vrothberg — COMMENTED** [review]
LGTM

File: libpod/stats.go [hunk 1]
```diff
@@ -91,22 +92,29 @@ func (c *Container) GetContainerStats(previousStats *define.ContainerStats) (*de
 	return stats, nil
 }
 
-// getMemory limit returns the memory limit for a given cgroup
-// If the configured memory limit is larger than the total memory on the sys, the
-// physical system memory size is returned
-func getMemLimit(cgroupLimit uint64) uint64 {
+// getMemory limit returns the memory limit for a container
+func (c *Container) getMemLimit() uint64 {
+	memLimit := uint64(math.MaxUint64)
```
Comment (mheon) [inline libpod/stats.go:97]: 99% sure this is a break from Docker's behavior - we should report a limit of 0 for containers that do not have a resource limit, not the maximum amount of RAM on the current system.

Comment (giuseppe) [reply libpod/stats.go:97]: I've changed it after a test was failing.

The current Podman behavior, AFAICS, is to report the maximum amount of RAM:

```
$ /usr/bin/podman run --rm -ti -d alpine top
58efadc49b77ef300eccb02715235616fb5729135198665f5ca7745006f93940
$ /usr/bin/podman stats -l --no-reset --no-stream
ID            NAME          CPU %       MEM USAGE / LIMIT  MEM %       NET IO      BLOCK IO    PIDS        CPU TIME    AVG CPU %
58efadc49b77  crazy_fermat  4.40%       233.5kB / 33.39GB  0.00%       -- / --     -- / --     1           7.2ms       4.40%
```

Comment (mheon) [reply libpod/stats.go:97]: Can you verify that `podman inspect` still reports 0 if not limited? If so, I'm fine with this.

Comment (giuseppe) [reply libpod/stats.go:97]: They return the same values:

```
$ diff <(bin/podman inspect -l | grep -i memory) <(/usr/bin/podman inspect -l | grep -i memory) | wc -l
0
```

**Pull request merged**